pkgs/stdenv/linux/default.nix

```diff
@@ -254,6 +254,19 @@ in
         '';
       });
 
+      gcc-unwrapped = self.stdenvNoCC.mkDerivation {
+        inherit (prevStage.gcc-unwrapped) name;
+        dontUnpack = true;
+        dontBuild = true;
+        installPhase = ''
+          mkdir -p $out
+          cp -ar ${prevStage.gcc-unwrapped}/* $out/
+          chmod -R a+w $out/
+          cp -a ${getLibc self}/lib/* $out/lib/
+          chmod -R a-w $out/
+        '';
+      };
+
```

Comment:
I encountered something similar in https://github.com/NixOS/nixpkgs/issues/158042 (when tried glibc-2.35 locally). I think the bug is in order or -L/-B options. I managed to get away with it by fixing cflags order: https://github.com/NixOS/nixpkgs/pull/158047

Reply:
The change seems to be a fix indeed, thanks!